visualization_tool/src/components/ControlMenu/ControlMenu.tsx

```diff
@@ -1,21 +1,104 @@
-import IconButton from '@mui/material/IconButton';
-import SearchIcon from '@mui/icons-material/Search';
+import {useState, useRef} from 'react';
+import AddIcon from '@mui/icons-material/Add';
+import CloseIcon from '@mui/icons-material/Close';
+
+import {Resource} from '../../types/resources';
+import {OutputFile} from '../../types/resources';
+
+import {parseData} from './Controller';
 
 import './ControlMenu.css';
 
-const ControlMenu = () => {
+type ControlMenuProps = {
+  setResources: React.Dispatch<React.SetStateAction<Resource[]>>;
+};
+
+const ControlMenu = ({setResources}: ControlMenuProps) => {
+  const fileInput = useRef<HTMLInputElement>(null);
+  const [files, setFiles] = useState<string[]>([]);
   return (
     <div className="control-menu">
-      <div className="search-bar">
-        <input
-          type="text"
-          name="search-input"
-          id="search-input"
-          placeholder="Resource Name"
-        />
-        <IconButton sx={{p: '0', color: '#3367D6'}} aria-label="search">
-          <SearchIcon />
-        </IconButton>
+      <div className="menu-item">
+        <div className="menu-item__header">
+          <span>
+            <img src="./icons/upload.png" alt="" />
+          </span>
+          <h3>Upload</h3>
+        </div>
+        <form
+          className="upload-form"
+          onSubmit={e => {
+            e.preventDefault();
+            const file = fileInput.current?.files?.[0];
+
+            if (!file) return;
+            // check if file is already uploaded
+            if (files.includes(file.name)) return;
+            const reader = new FileReader();
+            reader.readAsText(file);
+            reader.onload = e => {
+              const result = e.target?.result as string;
+              const data = JSON.parse(result) as OutputFile;
+
+              const resources = parseData(data, file.name);
```

Comment:
How are parsing error handled?

Reply:
Added a feedback error message and surrounded the parsing code with a try, catch block 

![Screenshot from 2023-07-14 19-42-23](https://github.com/google/gcp_scanner/assets/30212455/564839fd-82b9-4688-8bd6-dc0efd746d94)

In the future, we can add a function to validate the keys of the input JSON for more descriptive error message